sp_api/api/reports/reports.py

```diff
@@ -110,7 +110,11 @@ def get_report_document(self, document_id, decrypt: bool = False, file=None, **
                 'document': document
             })
             if file:
-                file.write(document)
+                if isinstance(file, str):
+                    with open(file, "w") as text_file:
+                        text_file.write(document)
+                else:
+                    file.write(document)
```

Comment:
@saleweaver 
`file` is a user-specified directory path right?  Would this still work if `file` was not a string or would it error?

Reply:
file usually can be a file or file like object. Since the reports usually are handled by some process, I use BytesIO and StringIO instead. I didn't have a single use case in 10 years that would merit actually saving a report as a txt file, only to read the same file by the next process.